### PR TITLE
set user name to sunetid, and populate another field with the user di…

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- set user name to sunetid, and populate another field with the user display name
- Do we need an update hook to create the fields? if the field doesn't exist, nothing happens.

# Needed By (Date)
- october

# Urgency
- low

# Steps to Test
1. check this out
1. delete your user
1. log back in with saml
1. verify you display name is populated in the field.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: config/install/field.field.user.user.field_user_name.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+langcode: en
+status: true
+dependencies:
+  config:
+    - field.storage.user.field_user_name
```
Comment (sherakama) [inline config/install/field.field.user.user.field_user_name.yml]: Perhaps change this field name to display_name as it initially confused me with the core user's `name` field.

File: stanford_ssp.module [hunk 2]
```diff
@@ -96,3 +97,18 @@ function stanford_ssp_simplesamlphp_auth_existing_user($attributes) {
...
 }
+
+/**
+ * Implements hook_simplesamlphp_auth_user_attributes().
+ *
+ * Set the display name field for the user from the attributes.
+ *
+ * @see https://www.drupal.org/node/2935428
+ */
+function stanford_ssp_simplesamlphp_auth_user_attributes(UserInterface $account, $attributes) {
```
Comment (sherakama) [inline stanford_ssp.module]: This hook gets called before the user entity gets saved?

**Comment by sherakama** [comment]
Do we need to change the display of the user entity to show the display name instead of the name field? @jbickar Do you think users would have an issue seeing their sunet id as their account insted of their full name?

**Comment by pookmish** [comment]
switching the name with the display name might best in profile/theme since it might involve some template or layout work.

File: stanford_ssp.module (hunk 2 above)
Comment (pookmish) [reply stanford_ssp.module]: It is called after the user entity is created and saved, after `externalauth.login` event, then it is re-saved if any changes have occur.

**Comment by jbickar** [comment]
Depends on LoE. I think SUnetID as display name is kind of janky but it is technically their unique ID.

**Comment by pookmish** [comment]
SUnetID is the "name", not display name. With this, it sets the SUnetID as the account name, and the display name gets populated into a field on the entity. Then a profile/theme can modify the entity to display the display name instead of the name property

**Comment by pookmish** [comment]
Thoughts??

**Comment by sherakama** [comment]
I like this change but I do think it will need upgrade path handling. Can we update previous versions to have the new field and data mappings?

**Comment by jbickar** [comment]
3 things based on conversation:

1. I think we want to namespace the field with `_s_` or _su_` or `_stanford_`
2. I think we do want an update hook if it will work. Looking at the things that make up the Unique ID, it should be OK.
3. Does `stanford_ssp_simplesamlphp_auth_user_attributes()` get run every time the user logs in? I think that's OK. If someone updates their name in StanfordWho, I think it's good to update in Drupal.

File: config/install/field.field.user.user.field_user_display_name.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+langcode: en
+status: true
+dependencies:
+  config:
+    - field.storage.user.field_user_display_name
```
Comment (sherakama) [inline config/install/field.field.user.user.field_user_display_name.yml]: new `su_` prefix.

**Comment by pookmish** [comment]
This is ready for another review. Fields have been renamed and an update hook has been added.

**Comment by pookmish** [comment]
poke merge?

**Comment by sherakama** [comment]
@boznik any comments before we merge?

**Comment by boznik** [comment]
i haven't reviewed this yet, but won't be able to get to it until this afternoon. is that soon enough?

**Comment by pookmish** [comment]
an extra day or two won't really matter. its just almost sitting at a 3 week old PR though.

**Pull request merged**